clients/pkg/promtail/wal/writer.go

```diff
@@ -27,14 +27,21 @@ const (
 	minimumCleanSegmentsEvery = time.Second
 )
 
-// WriterEventSubscriber is an interface that objects that want to receive events from the wal Writer can implement. After
-// they can subscribe to events by adding themselves as subscribers on the Writer with writer.Subscribe.
-type WriterEventSubscriber interface {
+// CleanupEventSubscriber is an interface that objects that want to receive events from the wal Writer can implement. After
+// they can subscribe to events by adding themselves as subscribers on the Writer with writer.SubscribeCleanup.
+type CleanupEventSubscriber interface {
 	// WriteCleanup is implemented by WriteTo, who can subscribe to these type of events for cleaning up the series cache
 	// entries that originated from deleted segments.
```

Comment:
yep, agree the commenet should go on the implementor side. Also, the name write to was inspired in the one picked by prometheus. Since in here the WriteTo has become something more of bits and pieces of actions the watcher performs, I'm going with `WatcherActions`. wdyt?

Reply:
I'll defer renaming all `WriteTo` stuff to `WatcherActions` once we agree on a name, since pushing that in this same PR will add a lot of noise

Reply:
> I think we're missing a way of removing the client/manager from the list of subscribers when it goes away?

Do we need such subscriber removal action implemented? The only time a subscriber, be it a watcher or a write to, are stopped, is when Promtail is stopped. This could happen when the config is re-loaded, or it's actually stopped. Since when re-loading the whole `client.Manager` and `wal.Writer` are re-created from scratch, we don't care about removing the subscribers.

Reply:
to me `WatcherActions` implies actions you can perform against the Watcher, I think keeping the name as `WriteTo` is okay the point was more that the comment here could become out of date if the interface name changed or the interface was split

if you want a better name, what about something like `WALDataReciever` or just `DataReceiver`? something that makes it clear that it's the interface that's used by whatever is sending data to the Client to be processed and sent on to Loki

Reply:
I prefer for now to stick with with `WriteTo`, since renaming that interface involves a LOT of renames over the `wal` package. Also, I like that is the same used in prometheus, which makes transfer of ideas from one project to the other easier.